src/resolve/resolvectl.c

```diff
@@ -1430,11 +1430,15 @@ static int status_ifindex(sd_bus *bus, int ifindex, const char *name, StatusMode
         if (mode == STATUS_DNS)
                 return status_print_strv_ifindex(ifindex, name, link_info.dns_ex ?: link_info.dns);
 
-        if (mode == STATUS_DOMAIN)
+        if (mode == STATUS_DOMAIN) {
+                strv_sort(link_info.domains);
                 return status_print_strv_ifindex(ifindex, name, link_info.domains);
+        }
 
-        if (mode == STATUS_NTA)
+        if (mode == STATUS_NTA) {
+                strv_sort(link_info.ntas);
```

Comment:
(maybe we should actually add a generic handler for bus_properties_map[] that not only parses strv, but also sorts it client side? might be useful elsewhere, and make this more beautiful, since the structure remains immutable after parsing)

Reply:
I did that now. It's less efficient in some sense because we sort and resort after each addition (not one item, but one strv_extend_strv() call). But strv_extend_strv() is already fairly inefficient, and we're unlikely to call this on very long data. So I think it's OK to be inefficient here.